Okta.Core/Clients/UsersClient.cs

```diff
@@ -51,6 +51,11 @@ public virtual User Get(string userId)
         /// <example>userClient.GetByUsername("user@domain.local") or userClient.GetByUsername("user")</example>
         public virtual User GetByUsername(string userName)
         {
+            if (!userName.StartsWith("?"))
+            {
+                return base.Get(userName);
+            }
+
             User user = null;
             var filter = new FilterBuilder();
             filter.Where("profile.login").EqualTo(userName);

```

Comment:
Instead of  `return user;`, let's actually do `return base.Get(user.Id);`

This will do two requests, but we'll be guaranteed the links.